Task/283/ensure all pipelines succeed

cyberbikepunk opened the pull request:
@akariv 

This processor is a replacement for the `stream_from_path` processor, whose functional approach became somewhat unreadable and brittle to changes. This new object-oriented approach should improve readability and facilitate the handling of special cases through sub-classing.

cyberbikepunk committed:
Add a new object oriented file ingestion processor.

cyberbikepunk committed:
Add chardet to the requirements.

cyberbikepunk committed:
Issue a log warning instead of failing when fiscal casting fails.

cyberbikepunk committed:
Replace the old ingestion processor with the new one.

cyberbikepunk committed:
Change the default ingestion processor to `ingest_local_file`.

akariv commented on requirements.txt:
```diff
@@ -1,3 +1,4 @@
+chardet
```
why not cchardet?
from our experience it provides better results (and much quickly)

cyberbikepunk replied:
thanks for the tip

cyberbikepunk commented:
@akariv I'm not merging this PR until I've written unit-tests.

akariv commented on common/processors/ingest_local_file.py:
```diff
@@ -0,0 +1,265 @@
+    """An ingestor for xls and xlsx files."""
+
+    @staticmethod
+    def force_strings(rows):
```
is someone using this method?
also:
what about XLSXIngestor?

cyberbikepunk replied:
I'm writing them as i need them. So far, I haven't had a use case. I'm not outright replacing the old module. I'm introducing it gradually for new and broken pipelines.

akariv reviewed (commented):
The code looks okay, although there's some overlap with what tabulator does.

cyberbikepunk commented:
@akariv @roll yes there is a lot of overlap with tabulator. It's a little unsatisfactory. But I turned the problem around and couldn't come up with a more integrated solution. For example: opening the file, cleaning the headers, and then re-opening to read the data is far from ideal.

cyberbikepunk committed:
Fix the XLSIngestor class.

- add `_post_processors` property to XLSIngestor
- rename `_options` to `_body_options`
- split `_header_options` from `_raw_headers`

cyberbikepunk committed:
Ingest BE2.vlaams.gewest.

- Fix the source description (1 column was wrong)
- Make a backup of the original data file
- Clean up the XLS file by hand (headers, footers)
- bootstrap the pipeline

pwalsh commented:
@cyberbikepunk @akariv @roll this basically does stuff tabulator does with much much less handling of corner cases. I can't see how it was better to put the time in writing this, than it is to learn the tabulator API and/or do some small PRs to it if there was something missing (based on what I see here, there is nothing that can't be done in tabulator).

roll commented:
@cyberbikepunk 
So what's challenges you've faced using processors pattern?

``` python
def processor(extended_rows):
    for number, headers, row in extended_rows:
        # clean headers
        # skip bad rows
        # ...
        # make keyed row
        yield keyed_row

def rows():
    return Stream(source, post_parse=[processor])
```

pwalsh commented:
@roll small typo :)

```
def rows():
    return Stream(source, post_parse=[processor])
```

cyberbikepunk commented:
Travis is not happy about changes in [os-types](https://github.com/openspending/os-types/commit/4e83e5a7b62a69b5efce77fcab1426b8969f46a7). See issue #297.
